### PR TITLE
Improve comment placement when refactoring code.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/17126

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpAsAndNullCheckTests.cs [hunk 1]
```diff
@@ -461,5 +461,49 @@ void M(object y)
     }
 }", compareTokens: false);
         }
+
+        [WorkItem(17129, "https://github.com/dotnet/roslyn/issues/17129")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestTrivia2()
```
Comment (jasonmalinowski) [inline src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpAsAndNullCheckTests.cs]: It's unclear to me which trivia you were expecting to be testing here...

File: src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions.cs [hunk 2]
```diff
@@ -1142,5 +1143,29 @@ public static ConditionalAccessExpressionSyntax GetInnerMostConditionalAccessExp
 
             return false;
         }
+
+        public static SyntaxNode MoveNonIndentationTriviaTo(
```
Comment (jasonmalinowski) [inline src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions.cs]: I admit I'm not a fan of this as an extension method, or might prefer it in the form .WithNonIndentationTrivia() with the parameters swapped...it's just confusing which direction it's going.

File: src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpAsAndNullCheckTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpAsAndNullCheckTests.cs]: It's really that the indentation looks appropriate at the end.  Essentially this test is explicitly stating that it cares about *all* trivia, and that if anything changes in that regard, it needs to be looked at. i.e. as opposed to 99.999% of the rest of the CodeAction tests that do not care to test what sort of end formatting they get.

File: src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions.cs (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions.cs]: Uh... sure.  I can do that.

**Comment by CyrusNajmabadi** [comment]
retest integration_prtest  please

**Pull request merged**